src/Console/Application.php

```diff
@@ -96,23 +98,6 @@ public function run(InputInterface $input = null, OutputInterface $output = null
         $xdebug = new XdebugHandler(self::INFECTION_PREFIX, '--ansi');
         $xdebug->check();
 
-        if (PHP_SAPI !== 'phpdbg'
-            && !$this->isXdebugLoaded
-            && !XdebugHandler::getSkippedVersion()
-            && !$input->hasParameterOption('--coverage', true)
-            && !$this->isInitialTestPhpOptionHasXdebug($input)
-        ) {
-            $this->io->error([
```

Comment:
We can set a flag in the containter  (here or in doRun)  intead of outputting an error here, and InfectionCommand could check it and err. This error text is also only important for the InfectionCommand, and only. Outputting this error is beyond resposibilities of the Application.

Reply:
make sense

Reply:
If the check comes later in the process then i think we can remove the `COMMAND_NAME` again, since we don't need to check it from the outside.

Reply:
Yeah, it is gone as I see